pkg/dns/azure/dns.go

```diff
@@ -85,20 +87,26 @@ func (m *provider) Ensure(record *iov1.DNSRecord, zone configv1.DNSZone) error {
 		return errors.Wrap(err, "failed to parse zoneID")
 	}
 
+	metadataLabel := m.config.InfraID
 	ARecordName, err := getARecordName(record.Spec.DNSName, targetZone.Name)
 	if err != nil {
 		return err
 	}
+	ARecord := client.ARecord{
+		Address: record.Spec.Targets[0],
+		Name:    ARecordName,
+		TTL:     record.Spec.RecordTTL,
+	}
+	if metadataLabel != "" {
+		ARecord.Label = fmt.Sprintf("kubernetes.io_cluster.%s", metadataLabel)
+	}
 
 	// TODO: handle >0 targets
 	err = m.client.Put(
 		context.TODO(),
 		*targetZone,
-		client.ARecord{
-			Address: record.Spec.Targets[0],
-			Name:    ARecordName,
-			TTL:     record.Spec.RecordTTL,
-		})
+		ARecord,
+	)
```

Comment:
Might as well unwrap the line now.  
```suggestion
	err = m.client.Put(context.TODO(), *targetZone, ARecord)
```